lib/annotations/sizes.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+/**
+ * Custom `@sizes` annotation. Expects sizes type followed by
+ * semicolon-separated list of `key: value` arguments.
+ */
+module.exports = () => ({
+  name: 'sizes',
+  multiple: false,
+  parse: raw => {
+    // expects e.g. 'color-palette; key: sans; sizes: text-sizes;'
+    // returns object {
+    //   type: "color-palette",
+    //   key: "sans",
+    //   sizes: "text-sizes"
+    // }
+    const options = {
+      type: 'sizes',
+    };
+    let key, value;
+    raw.split(';').forEach(option => {
+      const parts = option.split(':');
+      key = parts[0].trim();
+      value = parts[1] ? parts[1].trim() : null;
+      if (key) {
+        options[key] = value;
+      }
+    });
+    return options;
```

Comment:
Can we abstract these three cases into a utility fn for converting `key: value; key: value` into an object?